### PR TITLE
harden e2e test by adding healthCheck

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

... according to https://github.com/apache/skywalking-rust/pull/9#issuecomment-1013614810.

This is a draft PR and I submit it for running tests. I'll set it as ready for review after validation.

### CODE REVIEW

**Comment by tisonkun** [comment]
@Shikugawa @wu-sheng I think the tests is somehow stable and you can review that whether the change is valid.

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -22,4 +22,7 @@ jobs:
           pip3 install --upgrade pip
           pip3 install setuptools
           pip3 install -r requirements.txt
+          until curl http://127.0.0.1:8081/healthCheck; do
+            sleep 1
+          done
```
Comment (wu-sheng) [inline .github/workflows/e2e.yml:27]: @kezhenxu94 Is this a good practice?

Comment (kezhenxu94) [reply .github/workflows/e2e.yml:27]: > @kezhenxu94 Is this a good practice?

Ideally I’d put this check in the `run_e2e.py` so when running locally the healthiness is also checked.

Comment (tisonkun) [reply .github/workflows/e2e.yml:27]: @kezhenxu94 that sounds reasonable. Let me push a follow up for it.

**Review by kezhenxu94 — APPROVED** [review]

File: tests/e2e/run_e2e.py [hunk 2]
```diff
@@ -43,6 +43,22 @@ def validate(expected_file_name):
   
   args = parser.parse_args()
 
+  health_check_times = 0
```
Comment (Shikugawa) [inline tests/e2e/run_e2e.py:46]: Could it be implemented using Retry module? https://pypi.org/project/retry/

**Review by Shikugawa — COMMENTED** [review]
Overall LGTM. A nit comment.

**Comment by Shikugawa** [comment]
This PR is very helpful and worth doing to implement other agent implementations using the same mechanism on regression tests, such as cpp2sky.

**Comment by tisonkun** [comment]
@Shikugawa @wu-sheng this PR is ready for review.

**Review by Shikugawa — APPROVED** [review]
Thanks!

**Comment by tisonkun** [comment]
@Shikugawa thanks for your review! You may clean up the squash commit message the next time. I think most of them are meaningless. Or you can ask me to do the squash :P

**Pull request merged**